### PR TITLE
Able to view transaction list on the Wallet page

### PR DESCRIPTION
**Pull request by guanbinrui**
## Summary

This PR will let the wallet page to present a list of transactions owned by the specific wallet. It describes the Ethereum transactions in a union format to be able to merge variant transaction data format from any source and to accept any provider to be added in the future. There are two data source will be concerned at the moment of the PR was opening: zeroin and debank.

## TODOs

+ [x] Fetch transaction history from debank.
+ [x] Add transaction history UI.

### CODE REVIEW

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

File: packages/maskbook/src/extension/options-page/DashboardRouters/Settings.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { useRef } from 'react'
+import  { useRef } from 'react'
```
Comment (Jack-Works) [inline packages/maskbook/src/extension/options-page/DashboardRouters/Settings.tsx:1]: This is a format error

**Comment by Tedko** [comment]
if debank part is finished we can deliver this first @guanbinrui and add zerion as a plus.

File: packages/maskbook/src/extension/options-page/DashboardRouters/Wallets.tsx [hunk 2]
```diff
@@ -106,6 +107,10 @@ export default function DashboardWalletsRouter() {
     const tokens = useTrustedERC20TokensFromDB()
     const { value: detailedTokens, retry: retryDetailedTokens } = useAssetsDetailedCallback(tokens)
 
+    const a = useTransactions()
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/options-page/DashboardRouters/Wallets.tsx:110]: Does it useful?

Comment (lelenei) [reply packages/maskbook/src/extension/options-page/DashboardRouters/Wallets.tsx:110]: test! it not use

File: package.json [hunk 3]
```diff
@@ -47,6 +47,7 @@
     "@dimensiondev/holoflows-kit": "0.8.0-20201215084554-2f9e757",
     "@dimensiondev/kit": "0.0.0-20201126061915-b58b8f0",
     "@dimensiondev/matrix-js-sdk-type": "=1.0.0-20200731085604-dc2de92",
+    "@dimensiondev/metamask-extension-provider": "^3.0.0-20210125122006-86e9bae",
```
Comment (Jack-Works) [inline package.json:50]: Hello! Please rebase this and add this dependency to `/packages/maskbook/package.json`, thanks

File: packages/dashboard/src/locales/i18n_generated.d.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export function useDashboardI18N(): {}
```
Comment (Jack-Works) [inline packages/dashboard/src/locales/i18n_generated.d.ts:1]: Hold on, this file should not be committed to git. Did I write `.gitignore` wrong?

File: packages/dashboard/src/locales/i18n_generated.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+/* eslint-disable */
```
Comment (Jack-Works) [inline packages/dashboard/src/locales/i18n_generated.js:1]: This file should not be committed either

File: packages/maskbook/src/_locales/en/i18n_generated.d.ts [hunk 6]
```diff
@@ -0,0 +1,971 @@
+export function useMaskI18N(): {
```
Comment (Jack-Works) [inline packages/maskbook/src/_locales/en/i18n_generated.d.ts:1]: And this one also shouldn't be committed

File: packages/maskbook/src/_locales/en/i18n_generated.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+/* eslint-disable */
```
Comment (Jack-Works) [inline packages/maskbook/src/_locales/en/i18n_generated.js:1]: And this one also shouldn't be committed

File: packages/maskbook/src/extension/options-page/DashboardRouters/Wallets.tsx [hunk 8]
```diff
@@ -24,6 +24,7 @@ import { useAssetsDetailedCallback } from '../../../web3/hooks/useAssetsDetailed
 import { WalletContent } from '../DashboardComponents/WalletContent'
 import { EthereumStatusBar } from '../../../web3/UI/EthereumStatusBar'
 import { extendsTheme } from '../../../utils/theme'
+import { useTransactions } from '../../../plugins/Wallet/hooks/useTransactions'
```
Comment (Jack-Works) [inline packages/maskbook/src/extension/options-page/DashboardRouters/Wallets.tsx:27]: Is it unused?

File: packages/theme/src/locales/i18n_generated.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
+/* eslint-disable */
```
Comment (Jack-Works) [inline packages/theme/src/locales/i18n_generated.js:1]: And this one also shouldn't be committed

File: packages/theme/src/locales/i18n_generated.d.ts [hunk 10]
```diff
@@ -0,0 +1,3 @@
+export function useMaskThemeI18N(): {}
```
Comment (Jack-Works) [inline packages/theme/src/locales/i18n_generated.d.ts:1]: And this one also shouldn't be committed

**Comment by Tedko** [comment]
Any updates @lelenei @guanbinrui

**Comment by septs** [comment]
image

File: packages/maskbook/src/extension/options-page/DashboardComponents/TransactionList/Record.tsx [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+import classNames from 'classnames'
+
+const useStyles = makeStyles(() =>
+    createStyles({
+        root: {
+            display: 'flex',
+            alignItems: 'center',
+        },
+        direction: {
+            marginLeft: '4px',
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/options-page/DashboardComponents/TransactionList/Record.tsx:14]:
```suggestion
            marginLeft: theme.spacing(0.5),
```

File: packages/maskbook/src/plugins/Wallet/apis/zerion.ts [hunk 12]
```diff
@@ -0,0 +1,6 @@
+// const ZERION_API = 'wss://api-v4.zerion.io/socket.io/?api_token=Zerion.oSQAHALTonDN9HYZiYSX5k6vnm4GZNcM&EIO=3&transport=websocket'
+const ZERIO_API = 'wss://zerion-api-v4.mk.network'
+
+export function getTransactionList() {
+    return []
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Wallet/apis/zerion.ts:5]: 👀

**Pull request merged**